index.ts

```diff
@@ -0,0 +1,42 @@
+module browser_fs {
+    export class SourceFile {
+        fileName: string;
+        text: string = "";
+
+        constructor(_fileName: string) {
+            this.fileName = _fileName;
+        }
+    };
+
+    class ScriptTarget {};
+
+    export class FileSystem {
+        fileSystem: { [fileName: string]: SourceFile } = {};
+
+        getSourceFile(fileName: string, languageVersion: ScriptTarget, 
+                                        onError?: (message: string) => void): SourceFile
```

Comment:
I added onError and languageVersion to mimic the function prototype provided by the TypeScript CompilerHost interface. 
Will remove these, though.